### ISSUE TITLE
Main activity should be obtained automatically from Manifest file

### ISSUE THREAD

**Issue by AizazAZ** [Ereza/CustomActivityOnCrash#4]
CustomActivityOnCrash.setRestartActivityClass(MainActivity.class);
why not by default get Main Activity from Manifest file?

**Comment by Ereza** [comment]
That would be cool!

 There are some drawbacks however:
* An app may have more than one activity with intent filter with category LAUNCHER.
* An app may have NO launchable activity (for example, apps that are only widgets).

What should we do in those cases?

If you are willing to provide a proposal via pull request I can take a look at it.

Thanks!

**Comment by alorma** [comment]
Maybe it can add a custom action in order to be launched from library:

```xml
        <activity
            android:name=".view.activity.ErrorActivity"
            android:label="@string/app_name"
            android:theme="@style/EvaTheme.Drawer">
            <intent-filter>
                <action android:name="cat.ereza.customactivityoncrash.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
```

**Comment by alorma** [comment]
Library should check for this action first, if there is no Activity with that, search for default android launcher activities, and if not launch the default library error Activity

**Comment by Ereza** [comment]
@alorma I think that's more complicated than the currently needed configuration... Anyway, we are talking about the activity that gets launched when pressing the "restart" button, so launching the error activity is independent from that.

IMHO, it should be like this:
* If a restart activity has been specified with `CustomActivityOnCrash.setRestartActivityClass()`, use that.
* If not specified, get the launchable activities via `PackageManager`, if one or more exist, just grab the first one by default (although I don't personally like that).
* If no activities are launchable, just close the app.

Also, add an option to always close the app instead of restarting.

**Comment by alorma** [comment]
Oh, sorry i missunderstood the Issue :D i will move it to a new Issue.

Anyway, it can be used in a similar way, with an extra action.

```xml
<action android:name="cat.ereza.customactivityoncrash.RELAUNCH" />
```

**Comment by Ereza** [comment]
Closed with the pull request #6 by @AizazAZ. Thanks!
